azure-configurationservice/azure/configuration/azure_configuration_client_prep.py

```diff
@@ -0,0 +1,123 @@
+# ------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+
+"""
+The methods in this module preprocess the parameters for the methods in
+:class:`azure.configuration.AzureConfigurationClient` and
+:class:`azure.configuration.aio.AzureConfigurationClientSync`
+"""
+
+import re
+from requests.structures import CaseInsensitiveDict
+
+
+def prep_get_configuration_setting(key, **kwargs):
+    if not key:
+        raise ValueError("key is mandatory to get a ConfigurationSetting object")
+    return kwargs.get("headers")
+
+
+def prep_add_configuration_setting(configuration_setting, **kwargs):
+    if configuration_setting is None:
+        raise ValueError("Object configuration_setting can not be None")
+    if not configuration_setting.key:
+        raise ValueError("key is mandatory to add a new ConfigurationSetting object")
+
+    custom_headers = CaseInsensitiveDict(kwargs.get("headers"))
+    custom_headers["if-none-match"] = quote_etag("*")
+    return custom_headers
+
+
+def prep_update_configuration_setting(key, etag=None, **kwargs):
```

Comment:
If you have a common base class for both clients, these can be instance methods.

Reply:
Have a common base class now and put most code in the base class for sync and async